Sources/AppcuesKit/Presentation/Traits/Appcues/ExperienceWrapperSlideAnimator.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  ExperienceWrapperSlideAnimator.swift
+//  AppcuesKit
+//
+//  Created by James Ellis on 9/7/23.
+//  Copyright © 2023 Appcues. All rights reserved.
+//
+
+import UIKit
+
+@available(iOS 13.0, *)
+internal class ExperienceWrapperSlideAnimator: NSObject, UIViewControllerAnimatedTransitioning {
+
+    enum ModalTransitionType {
+        case presentation
+        case dismissal
+    }
+
+    var transitionType: ModalTransitionType = .presentation
+
+    private let view: ExperienceWrapperView
+    private let edgeIn: AppcuesModalTrait.TransitionEdge
+    private let edgeOut: AppcuesModalTrait.TransitionEdge
+
+    private var transitionDuration: TimeInterval {
+        switch transitionType {
+        case .presentation:
+            return 1.0
+        case .dismissal:
+            return 0.4
+        }
+    }
+
+    init(view: ExperienceWrapperView, edgeIn: AppcuesModalTrait.TransitionEdge, edgeOut: AppcuesModalTrait.TransitionEdge) {
+        self.view = view
+        self.edgeIn = edgeIn
+        self.edgeOut = edgeOut
+    }
+
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return transitionDuration
+    }
+
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+
+        let animator: UIViewPropertyAnimator
+        switch transitionType {
+        case .presentation:
+            animator = UIViewPropertyAnimator(duration: transitionDuration, dampingRatio: 0.82)
+        case .dismissal:
+            animator = UIViewPropertyAnimator(duration: transitionDuration, curve: UIView.AnimationCurve.easeIn)
+        }
+
+        switch transitionType {
+        case .presentation:
+            // We need to add the modal to the view hierarchy,
+            // and perform the animation.
+            if let toView = transitionContext.view(forKey: .to) {
+                toView.translatesAutoresizingMaskIntoConstraints = false
+                transitionContext.containerView.addSubview(toView)
+                toView.pin(to: transitionContext.containerView)
+                toView.layoutIfNeeded()
+            }
+
+            // move the modal off screen
+            UIView.performWithoutAnimation { beginTransition(transitionContext) }
+            // then slide it in
+            animator.addAnimations { [weak self] in self?.endTransition() }
+        case .dismissal:
+            // The modal is already in the view hierarchy,
+            // so we just perform the animation.
+            animator.addAnimations { [weak self] in self?.beginTransition(transitionContext) }
+        }
```

Comment:
noticing something on dismissal that may need more though. If we have multiple slideout groups in a sequence, in an experience, we end up getting a blinking backdrop effect, as the backdrop is not smoothly faded in and out like the normal default fade (crossDissolve), between container presentations.

Reply:
Here is how I propose dealing with this aspect https://github.com/appcues/appcues-ios-sdk/pull/451/commits/420d255755d3543b36c7be0b66843ecf9868a028 -- the `AppcuesModalTrait` will provide the `backdropView` reference and it will get passed through to the animator to fade in and out.

We _could_ rely on the fact it is at `subviews[0]` and just access it that way, but I'd rather be more explicit about it. With the backdrop fading in and out during the slide, a multi-step-group flow looks pretty nice - matching normal dialog behavior.

https://github.com/appcues/appcues-ios-sdk/assets/19266448/7d9a5004-b8c4-4099-a8a9-8cadb31d396f

Reply:
note: after further discussion, this was reworked again in https://github.com/appcues/appcues-ios-sdk/pull/451/commits/a3f7c110863c0be4f20f0cf425846259b81462a8 to change and allow the wrapper VC/view to create its own backdrop, so it is readily available at the point it is needed here now.